doc/read-the-docs-site/plutus-doc.cabal

```diff
@@ -70,7 +70,7 @@ executable doc-doctests
     , base               >=4.9      && <5
     , bytestring
     , containers
-    , flat               <0.5
+    , flat               ^>=0.6
```

Comment:
We need to be careful with this, we have to check carefully that upstream hasn't broken anything. Is this change necessary for 9.6?

Reply:
The onl version does not compile with `9.6` (probably due to `base` bounds).

Reply:
If we can get away with `allow-newer`ing the base bounds then I think I'd mildly prefer this, since this is a sensitive dependency and this will change the bound for all GHC versions atm.

Reply:
Would it not make sense to do one or both of the following?

* Write tests for `flat` and submit them upstream.
* Write tests for `flat` in the Plutus repo.

Reply:
I tried removing the version bump for `flat` and added `allow-newer: flat:*`. The result was:
```
src/Flat/Decoder/Strict.hs:240:19: error: [GHC-31891]
    • Illegal term-level use of the type constructor or class ‘TA.Array’
    • imported qualified from ‘Data.Text.Array’ at src/Flat/Decoder/Strict.hs:45:1-53
    • Perhaps use variable ‘array’ (line 239)
    • In the first argument of ‘T.Text’, namely ‘(TA.Array array)’
      In the first argument of ‘return’, namely
        ‘(T.Text (TA.Array array) 0 (lengthInBytes `div` 2))’
      In a stmt of a 'do' block:
        return (T.Text (TA.Array array) 0 (lengthInBytes `div` 2))
    |
240 |   return (T.Text (TA.Array array) 0 (lengthInBytes `div` 2))
    |                   ^^^^^^^^
```

Reply:
Ok I am on it.

Reply:
The test infrastructure this needs got broken a while ago so it might be a little while until we can test this. I am tentatively in favour of merging since I think it's very likely not broken.

Reply:
The tests are finished locally. 
Both plutus-master and plutus-master+flat0.6.patch pass the current mainnet script tests